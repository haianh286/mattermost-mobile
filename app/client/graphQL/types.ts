// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {CustomStatusDuration} from '@app/constants';

export type GQLQuery = {
	errors?: GQLError[];
    user?: Partial<GQLUser>;
    config?: ClientConfig;
    license?: ClientLicense;
    teamMembers: Array<Partial<GQLTeamMembership>>;
    channels?: Array<Partial<GQLChannel>>;
    channelsLeft?: Array<Partial<GQLChannel>>;
    channelMembers?: Array<Partial<GQLChannelMembership>>;
}

export type GQLError = {
	message: string;
	path: Array<string | number>;
}

export type GQLUser = {
    id: string;
	username: string;
	email: string;
	firstName: string;
	lastName: string;
	nickname: string;
	isBot: boolean;
	isGuest: boolean;
	isSystemAdmin: boolean;
	createAt: number;
	deleteAt: number;
	authService: string;
	customStatus: Partial<GQLUserCustomStatus>;
	status: Partial<GQLUserStatus>;
	props: UserProps;
	notifyProps: UserNotifyProps;
	lastPictureUpdateAt: number;
	locale: string;
	timezone: UserTimezone;
	position: string;
	roles: Array<Partial<GQLRole>>;
	preferences: Array<Partial<GQLPreference>>;
}

export const gqlToClientUser = (m: Partial<GQLUser>): UserProfile => {
    return {
        auth_service: m.authService || '',
        create_at: m.createAt || 0,
        delete_at: m.deleteAt || 0,
        email: m.email || '',
        first_name: m.firstName || '',
        id: m.id || '',
        is_bot: m.isBot || false,
        last_name: m.lastName || '',
        last_picture_update: m.lastPictureUpdateAt || 0,
        locale: m.locale || '',
        nickname: m.nickname || '',
        notify_props: m.notifyProps!,
        position: m.position || '',
        roles: m.roles?.map((v) => v.name!).join(',') || '',
        update_at: 0,
        username: m.username || '',
        auth_data: '',
        email_verified: true,
        props: m.props,
        status: m.status?.status || '',
        terms_of_service_create_at: 0,
        terms_of_service_id: '',
        timezone: m.timezone,
    };
};

export type GQLTeamMembership = {
    team: Partial<GQLTeam>;
	user: Partial<GQLUser>;
	roles: Array<Partial<GQLRole>>;
	deleteAt: number;
	schemeGuest: boolean;
	schemeUser: boolean;
	schemeAdmin: boolean;
	sidebarCategories: Array<Partial<GQLSidebarCategory>>;
}

export const gqlToClientTeamMembership = (m: Partial<GQLTeamMembership>): TeamMembership => {
    return {
        team_id: m.team?.id || '',
        delete_at: m.deleteAt || 0,
        roles: m.roles?.map((v) => v.name!).join(',') || '',
        user_id: m.user?.id || '',
        scheme_admin: m.schemeAdmin || false,
        scheme_user: m.schemeUser || false,
        mention_count: 0,
        msg_count: 0,
    };
};

export type GQLSidebarCategory = {
    id: string;
	sorting: CategorySorting;
	type: CategoryType;
	displayName: string;
	muted: boolean;
	collapsed: boolean;
	channelIds: string[];
}

export const gqlToClientSidebarCategory = (m: Partial<GQLSidebarCategory>, teamId: string): CategoryWithChannels => {
    return {
        channel_ids: m.channelIds || [],
        collapsed: m.collapsed || false,
        display_name: m.displayName || '',
        id: m.id || '',
        muted: m.muted || false,
        sort_order: 0,
        sorting: m.sorting || 'alpha',
        team_id: teamId,
        type: m.type || 'channels',
    };
};

export type GQLTeam = {
    id: string;
	displayName: string;
	name: string;
	description: string;
	email: string;
	type: TeamType;
	companyName: string;
	allowedDomains: string;
	inviteId: string;
}

export const gqlToClientTeam = (m: Partial<GQLTeam>): Team => {
    return {
        allow_open_invite: false,
        allowed_domains: m.allowedDomains || '',
        company_name: m.companyName || '',
        create_at: 0,
        delete_at: 0,
        description: m.description || '',
        display_name: m.displayName || '',
        email: m.email || '',
        group_constrained: false,
        id: m.id || '',
        invite_id: m.inviteId || '',
        last_team_icon_update: 0,
        name: m.name || '',
        scheme_id: '',
        type: m.type || 'I',
        update_at: 0,
    };
};

export type GQLUserCustomStatus = {
    emoji: string;
	text: string;
	duration: CustomStatusDuration;
	expiresAt: string;
}

export type GQLUserStatus = {
    status: string;
	manual: boolean;
	lastActivityAt: number;
	activeChannel: string;
	dndEndTime: number;
}

export type GQLPreference = {
    userId: string;
	category: string;
	name: string;
	value: string;
}

export const gqlToClientPreference = (m: Partial<GQLPreference>): PreferenceType => {
    return {
        category: m.category || '',
        name: m.name || '',
        user_id: m.userId || '',
        value: m.value || '',
    };
};

export type GQLChannelMembership = {
    channel: GQLChannel;
	user: GQLUser;
	roles: GQLRole[];
	lastViewedAt: number;
	msgCount: number;
	mentionCount: number;
	mentionCountRoot: number;
	notifyProps: ChannelNotifyProps;
	lastUpdateAt: number;
	schemeGuest: boolean;
	schemeUser: boolean;
	schemeAdmin: boolean;
	explicitRoles: string;
	cursor: string;
}

export const gqlToClientChannelMembership = (m: Partial<GQLChannelMembership>): ChannelMembership => {
    return {
        channel_id: m.channel?.id || '',
        last_update_at: m.lastUpdateAt || 0,
        last_viewed_at: m.lastViewedAt || 0,
        mention_count: m.mentionCount || 0,
        msg_count: m.msgCount || 0,
        notify_props: m.notifyProps || {},
        roles: m.roles?.map((r) => r.name).join(',') || '',
        user_id: m.user?.id || '',
        is_unread: false,
        last_post_at: 0,
        post_root_id: '',
        scheme_admin: m.schemeAdmin,
        scheme_user: m.schemeUser,
    };
};

export type GQLChannel = {
    id: string;
	createAt: number;
	updateAt: number;
	deleteAt: number;
	type: ChannelType;
	displayName: string;
	prettyDisplayName: string;
	name: string;
	header: string;
	purpose: string;
	creatorId: string;
	schemeId: string;
	team: Partial<GQLTeam>;
	cursor: string;
}

export const gqlToClientChannel = (m: Partial<GQLChannel>): Channel => {
    return {
        create_at: m.createAt || 0,
        creator_id: m.creatorId || '',
        delete_at: m.deleteAt || 0,
        display_name: m.displayName || '',
        extra_update_at: 0,
        group_constrained: false,
        header: m.header || '',
        id: m.id || '',
        last_post_at: 0,
        name: m.name || '',
        purpose: m.purpose || '',
        scheme_id: m.schemeId || '',
        shared: false,
        team_id: m.team?.id || '',
        total_msg_count: 0,
        type: m.type || 'O',
        update_at: m.updateAt || 0,
        fake: false,
        isCurrent: false,
        status: '',
        teammate_id: '',
    };
};

export type GQLRole = {
    id: string;
	name: string;
	permissions: string[];
	schemeManaged: boolean;
	builtIn: boolean;
}

export const gqlToClientRole = (m: Partial<GQLRole>): Role => {
    return {
        id: m.id || '',
        name: m.name || '',
        permissions: m.permissions || [],
        built_in: m.builtIn,
        scheme_managed: m.schemeManaged,
    };
};
