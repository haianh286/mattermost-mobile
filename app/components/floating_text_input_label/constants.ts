// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const BORDER_DEFAULT_WIDTH = 1;
export const BORDER_FOCUSED_WIDTH = 2;

export const INPUT_CONTAINER_VERTICAL_SPACING = 12;
export const INPUT_CONTAINER_HORIZONTAL_SPACING = 16;

export const DEFAULT_INPUT_CONTAINER_HEIGHT = 48;
export const MULTILINE_INPUT_CONTAINER_HEIGHT = 100;

export const ADORNMENT_TO_INPUT_SPACE = 8;

