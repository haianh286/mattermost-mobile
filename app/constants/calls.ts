// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {toMilliseconds} from '@utils/datetime';

const RefreshConfigMillis = toMilliseconds({minutes: 20});

const RequiredServer = {
    FULL_VERSION: '7.1.0',
    MAJOR_VERSION: 7,
    MIN_VERSION: 1,
    PATCH_VERSION: 0,
};

const PluginId = 'com.mattermost.calls';

export const REACTION_TIMEOUT = 10000;
export const REACTION_LIMIT = 20;

export default {RequiredServer, RefreshConfigMillis, PluginId, REACTION_TIMEOUT};
