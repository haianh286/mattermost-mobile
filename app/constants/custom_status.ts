// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const CUSTOM_STATUS_TEXT_CHARACTER_LIMIT = 100;
export const SET_CUSTOM_STATUS_FAILURE = 'set_custom_status_failure';
