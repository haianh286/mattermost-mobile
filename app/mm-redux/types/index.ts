// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import * as actions from './actions';
import * as channels from './channels';
import * as client4 from './client4';
import * as errors from './errors';
import * as emojis from './emojis';
import * as files from './files';
import * as general from './general';
import * as groups from './groups';
import * as integrations from './integrations';
import * as posts from './posts';
import * as preferences from './preferences';
import * as reactions from './reactions';
import * as requests from './requests';
import * as roles from './roles';
import * as search from './search';
import * as store from './store';
import * as teams from './teams';
import * as typing from './typing';
import * as utilities from './utilities';
import * as users from './users';
import * as bots from './bots';
import * as plugins from './plugins';
import * as config from './config';
import * as threads from './threads';

export {
    config,
    actions,
    bots,
    plugins,
    store,
    channels,
    errors,
    emojis,
    files,
    general,
    groups,
    typing,
    integrations,
    utilities,
    posts,
    preferences,
    roles,
    client4,
    search,
    teams,
    requests,
    reactions,
    users,
    threads,
};
