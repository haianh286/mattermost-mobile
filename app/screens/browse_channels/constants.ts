// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const PUBLIC = 'public';
export const SHARED = 'shared';
export const ARCHIVED = 'archived';
