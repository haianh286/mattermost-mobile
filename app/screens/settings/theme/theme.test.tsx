// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {fireEvent, render} from '@testing-library/react-native';

import React from 'react';

import {Preferences} from '@mm-redux/constants';
import {OsColorSchemeName} from '@mm-redux/types/general';
import {shallowWithIntl} from '@test/intl-test-helper';
import {ReduxIntlProvider} from '@test/testing_library';

import Theme from './theme';

describe('Theme', () => {
    // https://github.com/callstack/react-native-testing-library/issues/329#issuecomment-737307473
    jest.mock('react-native/Libraries/Components/Switch/Switch', () => {
        const mockComponent = require('react-native/jest/mockComponent');
        return mockComponent('react-native/Libraries/Components/Switch/Switch');
    });

    const lightPremade = Preferences.THEMES.default;
    const darkPremade = Preferences.THEMES.windows10;
    const lightCustom = {...Preferences.THEMES.default, type: 'custom'};
    const darkCustom = {...Preferences.THEMES.windows10, type: 'custom'};

    const toSetLightPremade = Preferences.THEMES.organization;
    const toSetDarkPremade = Preferences.THEMES.mattermostDark;
    const toSetLightCustom = {...Preferences.THEMES.organization, type: 'custom'};
    const toSetDarkCustom = {...Preferences.THEMES.mattermostDark, type: 'custom'};

    const initialPremadeThemes = [lightPremade, darkPremade];
    const initialCustomThemes = [lightCustom, darkCustom];

    const userId = 'dummyUserId';
    const teamId = 'dummyTeamId';
    const savePreferences = jest.fn();
    const baseProps = {
        theme: Preferences.THEMES.default,
        allowedThemes: Object.keys(Preferences.THEMES).map((key) => ({key, ...Preferences.THEMES[key]})),
        defaultLightTheme: Preferences.THEMES.default,
        lightTheme: Preferences.THEMES.default,
        darkTheme: Preferences.THEMES.windows10,
        userId,
        isThemeSyncWithOsAvailable: true,
        enableThemeSync: true,
        teamId,
        isLandscape: false,
        isTablet: false,
        allowCustomThemes: true,
        osColorScheme: 'light' as OsColorSchemeName,
        actions: {savePreferences},
    };

    test.each([
        {
            initialThemes: initialPremadeThemes,
            toSet: toSetLightPremade,
            expectedLight: toSetLightPremade,
        },
        {
            initialThemes: initialPremadeThemes,
            toSet: toSetLightCustom,
            expectedLight: toSetLightCustom,
        },
        {
            initialThemes: initialPremadeThemes,
            toSet: toSetDarkPremade,
            expectedLight: initialPremadeThemes[0],
            expectedDark: toSetDarkPremade,
        },
        {
            initialThemes: initialPremadeThemes,
            toSet: toSetDarkCustom,
            expectedLight: initialPremadeThemes[0],
            expectedDark: toSetDarkCustom,
        },
        {
            initialThemes: initialCustomThemes,
            toSet: toSetLightPremade,
            expectedLight: toSetLightPremade,
        },
        {
            initialThemes: initialCustomThemes,
            toSet: toSetLightCustom,
            expectedLight: toSetLightCustom,
        },
        {
            initialThemes: initialCustomThemes,
            toSet: toSetDarkPremade,
            expectedLight: initialCustomThemes[0],
            expectedDark: toSetDarkPremade,
        },
        {
            initialThemes: initialCustomThemes,
            toSet: toSetDarkCustom,
            expectedLight: initialCustomThemes[0],
            expectedDark: toSetDarkCustom,
        },
    ])(
        'saves proper preferences when toggling OS sync',
        ({initialThemes, toSet, expectedLight, expectedDark}) => {
            const {getByTestId, update} = render(
                <ReduxIntlProvider>
                    <Theme
                        {...baseProps}
                        lightTheme={initialThemes[0]}
                        darkTheme={initialThemes[1]}
                    />
                </ReduxIntlProvider>,
            );
            update(
                <ReduxIntlProvider>
                    <Theme
                        {...baseProps}
                        lightTheme={initialThemes[0]}
                        darkTheme={initialThemes[1]}
                        enableThemeSync={false}
                    />
                </ReduxIntlProvider>,
            );
            update(
                <ReduxIntlProvider>
                    <Theme
                        {...baseProps}
                        lightTheme={toSet}
                        darkTheme={initialThemes[1]}
                        enableThemeSync={false}
                    />
                </ReduxIntlProvider>,
            );

            fireEvent(getByTestId('os_sync.switch'), 'valueChange', true);
            const expectedPref = [
                {category: Preferences.CATEGORY_ENABLE_THEME_SYNC, name: teamId, user_id: userId, value: 'true'},
                {category: Preferences.CATEGORY_THEME, name: teamId, user_id: userId, value: JSON.stringify(expectedLight)},
            ];
            if (expectedDark) {
                expectedPref.push({category: Preferences.CATEGORY_THEME_DARK, name: teamId, user_id: userId, value: JSON.stringify(expectedDark)});
            }
            expect(savePreferences).toHaveBeenCalledWith(userId, expectedPref);
        },
    );

    test('Matches snapshot when OS sync on', () => {
        const wrapper = shallowWithIntl(<Theme {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('Matches snapshot when OS sync off', () => {
        const wrapper = shallowWithIntl(
            <Theme
                {...baseProps}
                enableThemeSync={false}
            />,
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
