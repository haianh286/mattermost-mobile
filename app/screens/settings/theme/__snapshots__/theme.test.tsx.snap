// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Matches snapshot when OS sync off 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="theme_settings.screen"
>
  <Connect(StatusBar) />
  <ScrollView
    alwaysBounceVertical={false}
    contentContainerStyle={
      Object {
        "paddingVertical": 35,
      }
    }
    style={
      Object {
        "backgroundColor": "rgba(61,60,64,0.06)",
        "flex": 1,
      }
    }
  >
    <OsSyncSection
      onToggle={[Function]}
      osColorScheme="light"
      selected={false}
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "colorScheme": "light",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBg": "#ffffff",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
    />
    <ThemeSection
      activeThemeType="Mattermost"
      allowCustomThemes={true}
      allowedThemes={
        Array [
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "colorScheme": "light",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "key": "default",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          },
          Object {
            "awayIndicator": "#dcbd4e",
            "buttonBg": "#1dacfc",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#f2f4f8",
            "centerChannelColor": "#333333",
            "codeTheme": "github",
            "colorScheme": "light",
            "dndIndicator": "#ff6a6a",
            "errorTextColor": "#a94442",
            "key": "organization",
            "linkColor": "#2f81b7",
            "mentionBg": "#fbfbfb",
            "mentionBj": "#fbfbfb",
            "mentionColor": "#2071f7",
            "mentionHighlightBg": "#f3e197",
            "mentionHighlightLink": "#2f81b7",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#7dbe00",
            "sidebarBg": "#2071a7",
            "sidebarHeaderBg": "#2f81b7",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#7ab0d6",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#136197",
            "sidebarUnreadText": "#ffffff",
            "type": "Organization",
          },
          Object {
            "awayIndicator": "#c1b966",
            "buttonBg": "#4cbba4",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#2f3e4e",
            "centerChannelColor": "#dddddd",
            "codeTheme": "solarized-dark",
            "colorScheme": "dark",
            "dndIndicator": "#e81023",
            "errorTextColor": "#ff6461",
            "key": "mattermostDark",
            "linkColor": "#a4ffeb",
            "mentionBg": "#b74a4a",
            "mentionBj": "#b74a4a",
            "mentionColor": "#ffffff",
            "mentionHighlightBg": "#984063",
            "mentionHighlightLink": "#a4ffeb",
            "newMessageSeparator": "#5de5da",
            "onlineIndicator": "#65dcc8",
            "sidebarBg": "#1b2c3e",
            "sidebarHeaderBg": "#1b2c3e",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#66b9a7",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4a5664",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost Dark",
          },
          Object {
            "awayIndicator": "#c1b966",
            "buttonBg": "#0177e7",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#1f1f1f",
            "centerChannelColor": "#dddddd",
            "codeTheme": "monokai",
            "colorScheme": "dark",
            "dndIndicator": "#e81023",
            "errorTextColor": "#ff6461",
            "key": "windows10",
            "linkColor": "#0d93ff",
            "mentionBg": "#0177e7",
            "mentionBj": "#0177e7",
            "mentionColor": "#ffffff",
            "mentionHighlightBg": "#784098",
            "mentionHighlightLink": "#a4ffeb",
            "newMessageSeparator": "#cc992d",
            "onlineIndicator": "#399fff",
            "sidebarBg": "#171717",
            "sidebarHeaderBg": "#1f1f1f",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#196caf",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#302e30",
            "sidebarUnreadText": "#ffffff",
            "type": "Windows Dark",
          },
        ]
      }
      customThemeAvailable={false}
      headerDefaultMessage="Choose a theme"
      headerId="user.settings.display.theme.chooseTheme"
      isLandscape={false}
      isTablet={false}
      onSelect={[Function]}
      testID="all_themes"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "colorScheme": "light",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBg": "#ffffff",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
    />
  </ScrollView>
</RNCSafeAreaView>
`;

exports[`Theme Matches snapshot when OS sync on 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
  testID="theme_settings.screen"
>
  <Connect(StatusBar) />
  <ScrollView
    alwaysBounceVertical={false}
    contentContainerStyle={
      Object {
        "paddingVertical": 35,
      }
    }
    style={
      Object {
        "backgroundColor": "rgba(61,60,64,0.06)",
        "flex": 1,
      }
    }
  >
    <OsSyncSection
      onToggle={[Function]}
      osColorScheme="light"
      selected={true}
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "colorScheme": "light",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBg": "#ffffff",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
    />
    <React.Fragment>
      <ThemeSection
        activeThemeType="Mattermost"
        allowCustomThemes={true}
        allowedThemes={
          Array [
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "colorScheme": "light",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "key": "default",
              "linkColor": "#2389d7",
              "mentionBg": "#ffffff",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            },
            Object {
              "awayIndicator": "#dcbd4e",
              "buttonBg": "#1dacfc",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#f2f4f8",
              "centerChannelColor": "#333333",
              "codeTheme": "github",
              "colorScheme": "light",
              "dndIndicator": "#ff6a6a",
              "errorTextColor": "#a94442",
              "key": "organization",
              "linkColor": "#2f81b7",
              "mentionBg": "#fbfbfb",
              "mentionBj": "#fbfbfb",
              "mentionColor": "#2071f7",
              "mentionHighlightBg": "#f3e197",
              "mentionHighlightLink": "#2f81b7",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#7dbe00",
              "sidebarBg": "#2071a7",
              "sidebarHeaderBg": "#2f81b7",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#7ab0d6",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#136197",
              "sidebarUnreadText": "#ffffff",
              "type": "Organization",
            },
          ]
        }
        customThemeAvailable={false}
        headerDefaultMessage="Choose a light theme"
        headerId="user.settings.display.theme.chooseLight"
        isLandscape={false}
        isTablet={false}
        onSelect={[Function]}
        testID="light_themes"
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "colorScheme": "light",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
      <ThemeSection
        activeThemeType="Windows Dark"
        allowCustomThemes={true}
        allowedThemes={
          Array [
            Object {
              "awayIndicator": "#c1b966",
              "buttonBg": "#4cbba4",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#2f3e4e",
              "centerChannelColor": "#dddddd",
              "codeTheme": "solarized-dark",
              "colorScheme": "dark",
              "dndIndicator": "#e81023",
              "errorTextColor": "#ff6461",
              "key": "mattermostDark",
              "linkColor": "#a4ffeb",
              "mentionBg": "#b74a4a",
              "mentionBj": "#b74a4a",
              "mentionColor": "#ffffff",
              "mentionHighlightBg": "#984063",
              "mentionHighlightLink": "#a4ffeb",
              "newMessageSeparator": "#5de5da",
              "onlineIndicator": "#65dcc8",
              "sidebarBg": "#1b2c3e",
              "sidebarHeaderBg": "#1b2c3e",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#66b9a7",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4a5664",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost Dark",
            },
            Object {
              "awayIndicator": "#c1b966",
              "buttonBg": "#0177e7",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#1f1f1f",
              "centerChannelColor": "#dddddd",
              "codeTheme": "monokai",
              "colorScheme": "dark",
              "dndIndicator": "#e81023",
              "errorTextColor": "#ff6461",
              "key": "windows10",
              "linkColor": "#0d93ff",
              "mentionBg": "#0177e7",
              "mentionBj": "#0177e7",
              "mentionColor": "#ffffff",
              "mentionHighlightBg": "#784098",
              "mentionHighlightLink": "#a4ffeb",
              "newMessageSeparator": "#cc992d",
              "onlineIndicator": "#399fff",
              "sidebarBg": "#171717",
              "sidebarHeaderBg": "#1f1f1f",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#196caf",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#302e30",
              "sidebarUnreadText": "#ffffff",
              "type": "Windows Dark",
            },
          ]
        }
        customThemeAvailable={false}
        headerDefaultMessage="Choose a dark theme"
        headerId="user.settings.display.theme.chooseDark"
        isLandscape={false}
        isTablet={false}
        onSelect={[Function]}
        testID="dark_themes"
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "colorScheme": "light",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
      />
    </React.Fragment>
  </ScrollView>
</RNCSafeAreaView>
`;
